API/src/core/ApiError.ts

```diff
@@ -0,0 +1,231 @@
+import { Response } from 'express';
+import { BaseApiErrorObject } from '../app/shared/dtos/base-api-response.dto';
+import {
+    ApiResponse,
+    AuthFailureResponse,
+    BadRequestResponse,
+    ErrorResponse,
+    ForbiddenReponse,
+    InternalErrorResponse,
+    NotFoundResponse,
+} from './ApiResponse';
+
+enum ErrorType {
+    BAD_TOKEN = 'BadTokenError',
+    TOKEN_EXPIRED = 'TokenExpiredError',
+    UNAUTHORIZED = 'AuthFailureError',
+    ACCESS_TOKEN = 'AccessTokenError',
+    INTERNAL = 'InternalError',
+    NOT_FOUND = 'NotFoundError',
+    NO_ENTRY = 'NoEntryError',
+    NO_DATA = 'NoDataError',
+    BAD_REQUEST = 'BadRequestError',
+    FORBIDDEN = 'ForbiddenError',
+}
+
+enum ErrorCode {
+    BAD_TOKEN = 100,
+    TOKEN_EXPIRED = 101,
+    UNAUTHORIZED = 102,
+    ACCESS_TOKEN = 103,
+    INTERNAL = 200,
+    NOT_FOUND = 300,
+    NO_ENTRY = 301,
+    NO_DATA = 303,
+    BAD_REQUEST = 304,
+    FORBIDDEN = 104,
+}
```

Comment:
I think that code is actually redundant for now since we don't have any means of authorization. I thought we would have use JWT for auth. I'll just delete them now since we don't use them yet.